sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -1039,6 +1039,83 @@ case class GpuArraysOverlap(left: Expression, right: Expression)
   }
 }
 
+case class GpuArrayRemove(left: Expression, right: Expression) extends GpuBinaryExpression {
+
+  override def dataType: DataType = left.dataType
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    // For every null entry in rhs, replace corresponding row in lhs with null
+    val lhsWithNull = withResource(rhs.getBase.isNull) { rhsIsNull =>
+      withResource(GpuScalar.from(null, dataType)) { nullList =>
+        rhsIsNull.ifElse(nullList, lhs.getBase)
+      }
+    }
+    // Repeat every entry in rhs n times where n is number of elements in corresponding row in lhs
+    val repeatedRhs = withResource(GpuScalar.from(0, DataTypes.IntegerType)) { zero =>
+      withResource(lhsWithNull.countElements) { counts =>
+        // Replace null count with 0
+        withResource(counts.replaceNulls(zero)) { noNullCounts =>
+          withResource(new Table(rhs.getBase)) { table =>
+            table.repeat(noNullCounts).getColumn(0)
+          }
+        }
+      }
+    }
+    withResource(lhsWithNull) { lhsWithNull =>
+      withResource(lhsWithNull.getListOffsetsView) { offSets =>
+        withResource(lhsWithNull.getChildColumnView(0)) { flattenLhs =>
+          withResource(repeatedRhs) { flattenRhs =>
+            // Element-wise comparison
+            val flattenBoolMask = flattenLhs.equalToNullAware(flattenRhs)
+            // NaN comparison for arrays of float or double type
+            val boolMaskWithNan = if (flattenLhs.getType == DType.FLOAT32 || 
```

Comment:
nit: could we split this method up into smaller helper methods. This is getting rather complicated and making is smaller would really help with readability.